contrib/building.md

```diff
@@ -0,0 +1,280 @@
+# Working With Realm-JS
+
+
+<!--ts-->
+  * [Install Instructions For Mac](#instructions-for-macos)
+  * [Hello World Projects To Practice](#test-projects)
+  * [Building Realm for React Native **Android**](#building-for-android)
+  * [Building Realm for React Native **iOS**](#ios)
+  * [Building Realm for **NodeJS**](#building-for-nodejs)
+
+<!--te-->
+
+
+## Pre-Requisites
+
+Clone [realm-js project](https://github.com/realm/realm-js) and install the required dependencies:
+
+* XCode 9.4+
+* [NodeJS](https://nodejs.org/en/) version 10 or later
+* Android SDK 23+
+* Android NDK 21.0
+  - Available via the SDK Manager on Android Studio **Tools > SDK Manager**.
+
+### Instructions for MacOS:
+
+```sh
+  #Install brew
+  bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+
+  #Install NVM, OpenJDK 8 and cocoapods
+  brew install nvm openjdk@8 cocoapods
+
+```
+
+Android Setup to build from the command-line:
+
+```sh
+# Before following this make sure java is available in the command line.
+# Location of your Android SDK
+export ANDROID_HOME=$HOME/Library/Android/sdk
+
+#To install NDK
+$ANDROID_HOME/tools/bin/sdkmanager --install "ndk;21.0.6113669"
+
+#Setup NDK
+export ANDROID_NDK_HOME=$ANDROID_HOME/ndk/21.0.6113669
+```
+
+### Optimisations
+
+##### JDK
+If you are a user of Android Studio you can reuse their embedded JDK instead, which save you from installing it yourself:  
+
+```sh
+# Add JAVA_HOME
+export JAVA_HOME="/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home"
+```
+
+
+##### CCache
+
+To improve compilation speed you can use [ccache](https://ccache.dev/), to install you should do:
+
+```sh
+brew install ccache
+```
+
+
+
+### Projects For Testing
+
+To test your changes on react native you can clone this sample project.
+
+```sh
+git clone https://github.com/cesarvr/react-native-realm sample-rn-project
+cd sample-rn-project
+npm install
+```
+
+To test your changes on NodeJS you can clone this example.
+
+```sh
+git clone https://github.com/cesarvr/hello-world-realm-js hello-sync
+cd hello-sync
+npm install
+```
+
+## Workflow Suggestion
```

Comment:
I would suggest splitting this file and link to this workflow instead, since it is a little involved and probably not what an average contributor would start with.